### PR TITLE
phase 3 clean-up, phase 4 done

### PR DESCRIPTION
**Pull request by raul-mates**

### CODE REVIEW

File: src/app/chart/chart.component.ts [hunk 1]
```diff
@@ -1,20 +1,36 @@
...
-      .getData()
+    this.updateChartData();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['data'] && !changes['data'].firstChange) {
+      this.updateChartData();
+    }
+  }
+
```
Comment (Clau-777) [inline src/app/chart/chart.component.ts]: Refactor **ChartComponent** to remove **OnInit** and **OnChanges** lifecycle hooks. Instead use a setter on the @Input() data property to directly trigger updateChartData method whenever 'data' changes. 
This approach simplifies the component and reduces lifecycle complexity.

File: src/app/pie-chart/pie-chart.component.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+  ngOnInit(): void {
+    this.updatePieChartData();
+  }
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['data'] && !changes['data'].firstChange) {
+      this.updatePieChartData();
+    }
+  }
```
Comment (Clau-777) [inline src/app/pie-chart/pie-chart.component.ts:30]: Use the same approche as the one from the ChartComponent  => refactor **PieChartComponent** to remove **OnInit** and **OnChanges** lifecycle hooks. Instead use a setter on the @Input() data property to directly trigger updateChartData method whenever 'data' changes.

File: src/app/chart/chart.component.ts [hunk 3]
```diff
@@ -1,47 +1,42 @@
...
         orderId: order.orderId,
         orderPieces: order.orderPieces,
         customerName: order.customerName,
-      }));
+      })) as OrderSummary[];
 
     this.chartData = this.getChartData();
   }
 
-  getChartData() {
```
Comment (Clau-777) [inline src/app/chart/chart.component.ts:28]: Please refactor this method introducing a ranges array to define the range boundaries. Then iterate over this array to generate each chart data entry dynamically. This approach will reduce code repetition, improve maintainability and make future extensions simpler.

File: src/app/chart/chart.component.ts [hunk 4]
```diff
@@ -1,47 +1,42 @@
...
+  getChartData(): ChartData[] {
```
Comment (Clau-777) [inline src/app/chart/chart.component.ts:36]: Please refactor this method introducing a ranges array to define the range boundaries. Then iterate over this array to generate each chart data entry dynamically. This approach will reduce code repetition, improve maintainability and make future extensions simpler.

File: src/app/app.component.html [hunk 5]
```diff
@@ -1,9 +1,39 @@
...
+          left: '0',
+          width: '100vw',
+          height: '100vh'
+        }
+      : {
+          'background-color': '#fff'
+        }
+  "
+>
+  <div
```
Comment (raluca-gug1) [inline src/app/app.component.html]: if this part only appears on the table component would be better to move it in the specific component

File: src/app/app.component.html [hunk 6]
```diff
@@ -1,9 +1,39 @@
-<dx-toolbar id="toolbar" style="margin-bottom: 21px">
-  <dxi-item
-    widget="dxButton"
-    [options]="buttonOptions"
-    location="after"
-  ></dxi-item>
-</dx-toolbar>
+<div
+  class="main-container"
```
Comment (raluca-gug1) [inline src/app/app.component.html:2]: here we can use ngClass and add the styling in the css file. we rarely define any styling in the html file, only as a last resort.
even better would be to move this styling in the corresponding component

File: src/app/drawer/drawer.service.ts [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+export class DrawerService {
+  isDrawerOpen = signal(false);
+  fullData!: TableDataInterface[];
+  tableData = signal(this.fullData);
+
+  constructor(public mockDataService: MockDataService) {
+    this.fullData = this.mockDataService.getData();
+  }
+
+  getTableData() {
```
Comment (raluca-gug1) [inline src/app/drawer/drawer.service.ts:18]: if you would like you can use rxjs to simulate a backend response: return of(this.fullData)
this way in the component we'll get an observable you can subscribe to and the scenario is more similar to a BE call

**Pull request merged**